app/models.py

```diff
@@ -2448,6 +2469,13 @@ def delete(cls, obj_id):
         if obj.is_sl_subdomain:
             DeletedSubdomain.create(domain=obj.domain)
 
+        from app import alias_utils
+
+        for alias in Alias.filter_by(custom_domain_id=obj_id):
```

Comment:
This should also be done only if the CustomDomain is an SL domain. If the alias belongs to a user domain is should be able to be re-created.